.github/workflows/integration.yml

```diff
@@ -0,0 +1,120 @@
+---
+name: Integration Test
+
+permissions:
+  pull-requests: read
+  contents: read
+  statuses: write
+
+# Running testing farm needs TF_API_KEY secret available inside the forked repo.
+# So the pull_request_target trigger has to be used in this case. To protect the
+# secrets this workflow has a PR sender permission checking at first job. Only
+# collaborator with repo write or admin permission can run this workflow.
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
```

Comment:
I noticed the test isn't actually running on this PR...did you test this manually in some way?

Or do we need to merge and then test on a new PR?

Reply:
> Or do we need to merge and then test on a new PR?

Correct. `pull_request_target` only runs in the pull request's target branch, and not the pull request's branch itself. This PR can't trigger this workflow included in this PR. Merge this PR first, then I'll send another PR to test CI and test case.